### PR TITLE
feat!: add events to in-memory provider

### PR DESCRIPTION
**Pull request by lukas-reining**
## This PR

Implements ConfigurationChanged event for the In-Memory Provider.

@moredip seen that you implemented the provider, hope it is okay to add the events to it.

### CODE REVIEW

**Comment by beeme1mr** [comment]
Why is this PR breaking?

**Comment by lukas-reining** [comment]
> Why is this PR breaking?

Would be the same argument to me as here @beeme1mr https://github.com/open-feature/js-sdk-contrib/pull/422#issuecomment-1608100672

File: libs/providers/in-memory/src/lib/in-memory-provider.ts [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
-  TypeMismatchError,
   StandardResolutionReasons,
-  GeneralError,
-  FlagValue,
+  TypeMismatchError,
 } from '@openfeature/js-sdk';
 import { FlagConfiguration } from './flag-configuration';
 
 export class InMemoryProvider implements Provider {
+  public readonly events = new OpenFeatureEventEmitter();
```
Comment (toddbaert) [inline libs/providers/in-memory/src/lib/in-memory-provider.ts:16]:
```suggestion
  readonly events = new OpenFeatureEventEmitter();
```

nit.

**Review by toddbaert — APPROVED** [review]
lgtm

**Review by toddbaert — APPROVED** [review]

**Comment by toddbaert** [comment]
Merging this since it seems like a good, straightforward change.

**Pull request merged**